blocks/embed/embed.js

```diff
@@ -50,7 +50,7 @@ const embedYoutube = (url, isLite) => {
       embed += url.search;
     }
     embedHTML = `<div style="left: 0; width: 100%; height: 0; position: relative; padding-bottom: 56.25%;">
-        <iframe src="https://www.youtube.com${vid ? `/embed/${vid}?rel=0&v=${vid}${suffix}` : embed}" style="border: 0; top: 0; left: 0; width: 100%; height: 100%; position: absolute;"
+        <iframe src="https://www.youtube-nocookie.com${vid ? `/embed/${vid}?rel=0&v=${vid}${suffix}&enablejsapi=1` : embed}" style="border: 0; top: 0; left: 0; width: 100%; height: 100%; position: absolute;"
```

Comment:
What's the reason for using `www.youtube-nocookie.com` here?  As I understand, the original problem was to avoid video from playing before cookie consent which can be achieved with your other change in this pr.

Reply:
Sure @sdmcraft . Apart from playing videos post cookie consent, the clicks of these videos are also not getting recorded by the Tag Managers. We did further investigation on this issue. Seems like there is a concept of youtube-nocookie & enablejsapi=1 in the embed URL in order to get those tracking working. 

To enable that tracking , just as a reference - https://measureschool.com/track-youtube-videos/

Reply:
Sorry @jindaliiita  - here is the URL which was referred - https://help.consentmanager.net/books/cmp/page/working-with-youtube